controlgetlineageABCdynamics.jl

```diff
@@ -2,21 +2,39 @@ using Printf
 using Dates
 using LogExpFunctions
 
+# This is a mapping betweene supported model names and an integer number representing that
+# model, to be used inside CUDA kernels where we couldn't use strings.  Another option could
+# be to use enums, which make life easier in CUDA kernels (no need to use hacks), but are
+# just slightly less nice for error handling (but strings aren't ideal either).

```

Comment:
As a quick note for the future, probably [enums](https://docs.julialang.org/en/v1/base/base/#Base.Enums.@enum) (or [enumerated types](https://en.wikipedia.org/wiki/Enumerated_type)) would have been a good alternative (probably better for the use in CUDA kernels), but I didn't make that change and preferred to stick to strings as previously agreed, as they do the job anyway.

However, in case of a major rewrite of the code in the future I'd suggest using proper datatypes, something along the lines of
```julia
abstract type AbstractModel end
abstract type AbstractFrechetWeibull <: AbstractModel end
struct SimpleFrechetWeibull <: AbstractFrechetWeibull end
struct ClockFrechetWeibull <: AbstractFrechetWeibull end
struct RwFrechetWeibull <: AbstractFrechetWeibull end
struct TwoDRwFrechetWeibull <: AbstractFrechetWeibull end
abstract type AbstractGammaExponetial <: AbstractModel end
struct SimpleGammaExponetial <: AbstractGammaExponetial end
struct ClockGammaExponetial <: AbstractGammaExponetial end
struct RwGammaExponetial <: AbstractGammaExponetial end
struct TwoGammaExponetial <: AbstractGammaExponetial end
```
And then you can dispatch on these types and ***greatly*** simplify the logic of the code in multiple places, instead of having endless `if` conditions repeated multiple times.  This would also make it much easier to extend the code to other models by adding few methods for a new datatype, instead of making changes all over the place in existing functions.  Again, I couldn't make this change because of lack of time for such a major refactoring and no way to make sure the current functionalities would be preserved, but I figured I'd suggest this alternative strategy for the future.